subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/http/HttpAuthenticationDependencyResolutionIntegrationTest.groovy

```diff
@@ -162,6 +164,48 @@ task listJars {
         'basic and digest' | 'authentication { basic(BasicAuthentication)\ndigest(DigestAuthentication) }' | DIGEST            | ['Basic', 'Digest']
     }
 
+    @Unroll
+    @Issue("gradle/gradle#5571")
+    public void "can resolve dependencies using #authSchemeName scheme from #authScheme authenticated with http header HTTP maven repository"() {
+        given:
+        def moduleA = mavenHttpRepo.module('group', 'projectA', '1.2').publish()
+        and:
+        buildFile << """
+repositories {
+    maven {
+        url "${mavenHttpRepo.uri}"
+        credentials(HttpHeaderCredentials) { header 'TestHttpHeaderName: TestHttpHeaderValue' }
+        authentication { header(HttpHeaderAuthentication) }
+    }
+}
+configurations { compile }
+dependencies {
+    compile 'group:projectA:1.2'
+}
+task listJars {
+    doLast {
+        assert configurations.compile.collect { it.name } == ['projectA-1.2.jar']
+    }
+}
+"""
+
+        when:
+        serverAuthScheme = authScheme
+
+        and:
+        moduleA.pom.expectGet()
+        moduleA.artifact.expectGet()
+
+        then:
+        succeeds('listJars')
+        and:
+        server.authenticationAttempts.asList() == authenticationAttempts
+
+        where:
+        authSchemeName     | authScheme        | authenticationAttempts
+        'header'           | HEADER            | ['TestHttpHeaderValue']
```

Comment:
Please remove the `where` and inline all values.

Reply:
Done.